packages/components/src/internal/components/lineage/LineageSettings.tsx

```diff
@@ -0,0 +1,216 @@
+                            <li>
+                                <b>All</b> - Include all nodes from the seed available in the lineage response.
+                            </li>
+                            <li>
+                                <b>Multi</b> - Include all nodes from the seed until a depth is found that contains
+                                multiple nodes..
+                            </li>
+                            <li>
+                                <b>Nearest</b> - Include only the immediately connected nodes from the seed.
+                            </li>
+                            <li>
+                                <b>Specific</b> - Include all nodes from the seed up to the "Parent Generations" or
+                                "Child Generations" specified.
+                            </li>
+                        </ul>
+                    </LabelHelpTip>
+                </label>
+                <SelectInput
+                    clearable={false}
+                    name="generations"
+                    onChange={onGenerationChange}
+                    options={generations}
+                    value={options.grouping?.generations ?? DEFAULT_GROUPING_OPTIONS.generations}
+                />
+
+                <label>
+                    Child Generations
+                    <LabelHelpTip placement="top" title="Child Generations">
+                        <div className="lineage-settings__tip-body">
+                            The number of descendant generations to render from the seed node. Only applies when
+                            Generations is set to "Specific".
+                            <div>
+                                <b>Note:</b> A maximum of {DEFAULT_GROUPING_OPTIONS.childDepth * 2} generations are
+                                requested from the server.
+                            </div>
+                        </div>
+                    </LabelHelpTip>
+                </label>
+                <input
+                    defaultValue={options.grouping?.childDepth ?? DEFAULT_GROUPING_OPTIONS.childDepth}
+                    className="form-control"
+                    max={DEFAULT_GROUPING_OPTIONS.childDepth * 2}
+                    min={0}
+                    name="childDepth"
+                    onChange={onGroupingChange}
+                    type="number"
+                />
+
+                <label>
+                    Parent Generations
+                    <LabelHelpTip placement="top" title="Parent Generations">
+                        <div className="lineage-settings__tip-body">
+                            The number of ancestor generations to render from the seed node. Only applies when
+                            Generations is set to "Specific".
+                            <div>
+                                <b>Note:</b> A maximum of {DEFAULT_GROUPING_OPTIONS.childDepth * 2} generations are
```

Comment:
should this use `DEFAULT_GROUPING_OPTIONS.parentDepth * 2`?

Reply:
Revised.